addons/account/models/account_move.py

```diff
@@ -1497,7 +1497,7 @@ def _get_reconciled_info_JSON_values(self):
                 'payment_id': counterpart_line.id,
                 'partial_id': partial.id,
                 'account_payment_id': counterpart_line.payment_id.id,
-                'payment_method_name': counterpart_line.payment_id.payment_method_id.name if counterpart_line.journal_id.type == 'bank' else None,
+                'payment_method_name': counterpart_line.payment_id.payment_method_line_id.name if counterpart_line.journal_id.type == 'bank' else None,
```

Comment:
`counterpart_line.payment_id.payment_method_line_id.name` is enough. A payment can only be on a bank journal :)